### PR TITLE
Feat/compoundmetric

### LINKED ISSUE #35: CompoundMetric
Implement an abstract class CompoundMetricManager for computing metrics for a set of samples from a population.

Implement in conjunction with #36.

Requires #8, #28

### LINKED ISSUE #36: CompoundMetricManager
Implement a class for managing a set of CompoundMetrics.

Implement in conjunction with #35.

Requires #28

### LINKED ISSUE #40: Alpha-Precision CompoundMetric
Implement a CompoundMetric for computing alpha-precision, using a SampleMetricManager.

Make sure to append to  #44  if applicable.

#35

### PR DESCRIPTION
**Pull request by tediiieee**
In addition to the metric related commits, i added a small commit regarding the training of decoder (ASAD).

Note that the circular import introduced some problem with typing, so if a type is missing that's probably why.

Closes #36 
Closes #35

### CODE REVIEW

File: src/metric/CompoundMetric.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        """
+        pass
+
+    @abc.abstractmethod
+    def get(self, calc_if_missing: bool = False, **parameters: Any) -> Any:
+        """
+        Get the metric.
+
+        Args:
+            calc_if_missing (bool, optional): If True calculates the metric if missing
```
Comment (sporreking) [inline src/metric/CompoundMetric.py:103]: Should calculate*

File: src/metric/CompoundMetric.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                by calling `calc()`. Defaults to False.
+            **parameters (Any): Arbitrary parameters required by the metrics.
+                These are only used if `calc_if_missing=True`, and are all forwarded
+                to `calc()`.
+
+        Returns:
+            Any: Return the value of the metric. Should return None if the metric is missing
```
Comment (sporreking) [inline src/metric/CompoundMetric.py:110]: Redundant "Return...", start with "The..."

**Review by sporreking — CHANGES_REQUESTED** [review]
There were a couple of tiny things, decide yourself if they are worth changing

Looks very nice overall! Nice commit usage

File: src/metric/CompoundMetric.py (hunk 1 above)
Comment (tediiieee) [reply src/metric/CompoundMetric.py]: agree

File: src/metric/CompoundMetric.py (hunk 2 above)
Comment (tediiieee) [reply src/metric/CompoundMetric.py]: agree

**Review by sporreking — APPROVED** [review]
🥇

**Pull request merged**